game-logic.js

```diff
@@ -0,0 +1,367 @@
+// https://github.com/rocketacademy/basics-beat-that
+
+// 1. There are TWO players and players take turns.
+// 2. Each player roll TWO dice
+
+const PLAYER_ONE = 0;
+const PLAYER_TWO = 1;
+const LAST_PLAYER = PLAYER_TWO;
+const PLAYERS = 2;
+var CURRENT_PLAYER = 0;
+
+const PLAYER_WIN_COUNT = [0, 0];
+/**
+ * playerRolls
+ * An example outcome when both players throw dice
+ * playerRolls =  [
+ *                   [[1, 3]],
+ *                  [[6, 6]]
+ *                ];
+ */
+var playerRolls = [[], []];
+
+/**
+ * playersRollsOrderings
+ * An example outcome when both players throw dice
+ * playersRollsOrderings =   [
+ *                            [[1, 3],[3, 1]],
+ *                            [[6, 6]]
+ *                          ];
+ */
+var playersRollsOrderings = [[], []];
+
+/**
+ * playerOrderingChoice
+ * A example outcome when both players had chosen their ordering
+ * playerOrderingChoice =   [
+ *                            [[3, 1]],
+ *                            [[6, 6]]
+ *                          ];
+ */
+var playerOrderingChoice = [[], []];
+
+const ACTION_ROLL = 0;
+const ACTION_ORDER = 1;
+const ACTION_GAME_SET = 2;
+
+var CURRENT_ACTION = ACTION_ROLL;
+
+var PREV_ROUND_RESULT_DESCRIPTION = "";
+
+/** State Transition Functions */
+const resetGame = function () {
+  CURRENT_PLAYER = PLAYER_ONE;
+  playerRolls = [[], []];
+  playersRollsOrderings = [[], []];
+  playerOrderingChoice = [[], []];
+  PREV_ROUND_RESULT_DESCRIPTION = "";
+  return "Game reset";
+};
+const changeAction = function (ACTION) {
+  CURRENT_ACTION = ACTION;
+};
+const changePlayer = function () {
+  CURRENT_PLAYER = (CURRENT_PLAYER + 1) % PLAYERS;
+};
```

Comment:
these two are nice but i question the value of having these fairly simple one-line assignments as helper functions, seems overly granular to me but i can def see the benefits